File: doc-examples/hibernate-example-java/src/main/java/example/AccountRepository.java
package example;

import io.micronaut.data.annotation.Repository;
import io.micronaut.data.repository.CrudRepository;

import jakarta.validation.Valid;
import jakarta.validation.constraints.Min;

@Repository
public interface AccountRepository extends CrudRepository<@Valid Account, @Min(0) Long> {
}

File: doc-examples/hibernate-example-python/src/main/python/example/AccountRepository.py
from typing import Annotated

from jakarta.validation import Valid
from jakarta.validation.constraints import Min
from micronaut.data.annotation import Repository
from micronaut.data.repository import CrudRepository

from example.Account import Account


@Repository
class AccountRepository(CrudRepository[Annotated[Account, Valid], Annotated[int, Min(0)]]):
    pass
